src/test/java/org/olabs/portal/api/CollectionResourceIntTests.java

```diff
@@ -0,0 +1,589 @@
+  public void testGet_projection() throws IOException {
+
+    // query with projection
+    final JSONArray projected =
+        client.getJsonArray(
+            getCollectionUrl(TEST_GET_COLLECTION)
+                .query("f", new BasicDBObject().append("date", 1).append("variable", 1).toJson())
+                .toString());
+    assertEquals(3, projected.length());
+    projected.forEach(
+        o -> {
+          assertEquals(3, ((JSONObject) o).keySet().size());
+          assertNotNull(((JSONObject) o).get("variable"));
+          assertNotNull(((JSONObject) o).get("_id"));
+          final JSONObject dateVal = (JSONObject) ((JSONObject) o).get("date");
+          assertNotNull(dateVal);
+          assertEquals(MongoUtils.toISODateString(TEST_DATE), dateVal.get("$date"));
+        });
+  }
+
+  @Test
+  public void testGet_sort() throws IOException {
+    // query with sort
+    final JSONArray sorted =
+        client.getJsonArray(
+            getCollectionUrl(TEST_GET_COLLECTION)
+                .query("s", new BasicDBObject("variable", 1))
+                .toString());
+    assertEquals(3, sorted.length());
+    assertEquals(
+        List.of("1", "2", "3"),
+        sorted.toList().stream().map(o -> ((Map) o).get("variable")).collect(Collectors.toList()));
+    final JSONArray reversed =
+        client.getJsonArray(
+            getCollectionUrl(TEST_GET_COLLECTION)
+                .query("s", new BasicDBObject("variable", -1))
+                .toString());
+    assertEquals(3, reversed.length());
+    assertEquals(
+        List.of("3", "2", "1"),
+        reversed.toList().stream()
+            .map(o -> ((Map) o).get("variable"))
+            .collect(Collectors.toList()));
+  }
+
+  @Test
+  public void testGet_count() throws IOException {
+    // count
+    final String count =
+        client.get(getCollectionUrl(TEST_GET_COLLECTION).query("c", "true").toString());
+    assertEquals("3", count);
+    final String filteredCount =
+        client.get(
+            getCollectionUrl(TEST_GET_COLLECTION)
+                .query("c", "true")
+                .query("q", new BasicDBObject("variable", "1").toJson())
+                .toString());
+    assertEquals("1", filteredCount);
+  }
+
+  @Test
+  public void testGet_findOne() throws IOException {
+    // findOne
+    final JSONObject one =
+        client.getJson(getCollectionUrl(TEST_GET_COLLECTION).query("fo", "true").toString());
+    assertNotNull(one);
+    assertEquals(13, one.length());
+  }
+
+  @Test
+  public void testGet_findOne_sorted() throws IOException {
+    // this only works correctly in new API
+    assumeTrue(isTestClient());
+    final JSONObject oneSorted =
+        client.getJson(
+            getCollectionUrl(TEST_GET_COLLECTION)
+                .query("fo", "true")
+                .query("s", new BasicDBObject("variable", -1))
+                .toString());
+    assertNotNull(oneSorted);
+    assertEquals("3", oneSorted.get("variable"));
+  }
```

Comment:
Something interesting I discovered is that the production API does not honor the sort parameter when you do a findOne.  Seemed reasonable that the new API could fix that.